akka-stream-tests/src/test/scala/akka/stream/io/compression/CoderSpec.scala

```diff
@@ -0,0 +1,194 @@
+      val compressed = comp.compressAndFinish(inputBytes)
+
+      ourDecode(compressed) should equal(inputBytes)
+    }
+
+    "shouldn't produce huge ByteStrings for some input" in {
+      val array = new Array[Byte](10) // FIXME
+      util.Arrays.fill(array, 1.toByte)
```

Comment:
I just copied those over from akka-http but it would make sense, of course, to fix more things in the process.